packages/jaeger-ui/src/model/link-patterns.js

```diff
@@ -0,0 +1,183 @@
+// Copyright (c) 2017 The Jaeger Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import _uniq from 'lodash/uniq';
+import { getConfigValue } from '../utils/config/get-config';
+
+const parameterRegExp = /\$\{([^{}]*)\}/;
+
+export function processTemplate(template, encodeFn) {
+  if (typeof template !== 'string') {
+    if (!template || !Array.isArray(template.parameters) || !(template.template instanceof Function)) {
+      throw new Error('Invalid template');
+    }
+    return template;
+  }
+  const templateSplit = template.split(parameterRegExp);
+  const templateSplitLength = templateSplit.length;
+  const parameters = [];
+  // odd indexes contain variable names
+  for (let i = 1; i < templateSplitLength; i += 2) {
+    const param = templateSplit[i];
+    let paramIndex = parameters.indexOf(param);
+    if (paramIndex === -1) {
+      paramIndex = parameters.length;
+      parameters.push(param);
+    }
+    templateSplit[i] = paramIndex;
+  }
+  return {
+    parameters,
+    template: (...args) => {
+      let text = '';
+      for (let i = 0; i < templateSplitLength; i++) {
+        if (i % 2 === 0) {
+          text += templateSplit[i];
+        } else {
+          text += encodeFn(args[templateSplit[i]]);
+        }
+      }
+      return text;
+    },
+  };
+}
+
+export function createTestFunction(entry) {
+  if (typeof entry === 'string') {
+    return arg => arg === entry;
+  }
+  if (Array.isArray(entry)) {
+    return arg => entry.indexOf(arg) > -1;
+  }
+  if (entry instanceof RegExp) {
```

Comment:
Planning ahead in preparation for the JS config file (#123) is awesome but it's tough to validate functionality that relies on or is impacted by the JS config before it's implemented. That increases the risks when switching to using the JS config. Maybe the regular expression and function variants of the test function should be kept on ice until a PR for #128 is merged?

Reply:
I have now commented the parts of the code that rely on #123.